py/desitarget/secondary.py

```diff
@@ -494,6 +494,38 @@ def match_secondary(primtargs, scxdir, scndout, sep=1.,
     # ADM rename the SCND_TARGET column, in case this is an SV file.
     targs = rfn.rename_fields(targs, {'SCND_TARGET': desicols[3]})
 
+    # APC Secondary target bits only affect PRIORITY, NUMOBS and
+    # APC obsconditions for specific DESI_TARGET bits
+    # APC See https://github.com/desihub/desitarget/pull/530
+    from desitarget.targetmask import desi_mask
+    # APC Only consider primary targets with secondary bits set
+    scnd_update = (targs[desicols[0]] & desi_mask['SCND_ANY']) != 0
+    if np.any(scnd_update):
+        # APC Allow changes to primaries if the DESI_TARGET bitmask has
+        # APC only the following bits set, in any combination.
+        update_from_scnd_bits = desi_mask['MWS_ANY'] & desi_mask['STD_BRIGHT'] & desi_mask['STD_FAINT'] & desi_mask['STD_WD']
+        scnd_update &= ((targs[desicols[0]] & ~update_from_scnd_bits) == 0)
+        log.info('Setting new priority, numobs and obsconditions from secondary for {} matched primaries'.format(scnd_update.sum()))
+
+        # APC Primary and secondary obsconditions are or'd
+        scnd_obscon = set_obsconditions(targs[scnd_update], scnd=True)
+        targs['OBSCONDITIONS'][scnd_update] &= scnd_obscon
+
+        # APC bit of a hack here
```

Comment:
Wasn't sure if this test was necessary, or if this was the best way to do it. At least seems harmless.